jenkins-integration/Gemfile

```diff
@@ -2,3 +2,7 @@ source ENV['GEM_SOURCE'] || 'https://rubygems.org'
 
 gem 'beaker', '2.15.1'
 gem 'scooter', :git => 'git@github.com:puppetlabs/scooter.git', :tag => '2.2'
+
+if File.exists? "#{__FILE__}.local"
+  eval(File.read("#{__FILE__}.local"), binding)
+end
```

Comment:
FWIW, we do this in Puppet Server, too - https://github.com/puppetlabs/puppet-server/blob/puppet-server-2.1.1/Gemfile#L26-L28.  @jeffmccune probably could give more input here as to when this pattern was established in other Puppet projects.